tests/system/src/test/java/test/util/Util.java

```diff
@@ -349,8 +347,27 @@ public static <T extends Application> void launch (
     }
 
     /**
-     * This synchronous method first hides all the specified stages (ignoring any null Stages)
-     * in the platform thread, then calls {@link Platform.exit()}.
+     * Starts the JavaFX runtime, invoking the specified Runnable on the JavaFX application thread.
+     * This Runnable must call {@link CountdownLatch#countDown()} once to signal
+     * its successful start, otherwise an exception will be thrown when no such signal is received
+     * within 15 seconds.
+     *
+     * @param startupLatch - a latch used to communicate successful start of the application
+     * @param r - code to invoke on the application thread.
+     */
+    public static void startup(CountDownLatch startupLatch, Runnable r) {
+        Platform.startup(r);
+        try {
+            String msg = "Timeout waiting for FX runtime to start";
+            Assert.assertTrue(msg, startupLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
```

Comment:
I recommend to standardize on 15 seconds like you did for launch (the ones you replaced were mostly a mix of 10 or 15 seconds). Maybe you could create a STARTUP_TIMEOUT constant field?